op-service/sources/types.go

```diff
@@ -125,6 +125,11 @@ type rpcHeader struct {
 // checkPostMerge checks that the block header meets all criteria to be a valid ExecutionPayloadHeader,
 // see EIP-3675 (block header changes) and EIP-4399 (mixHash usage for prev-randao)
 func (hdr *rpcHeader) checkPostMerge() error {
+	// Skip checking if a block is post-merge for genesis blocks.
+	// We need this when validating the genesis block for op-goerli & op-mainnet in CheckL2GenesisBlockHash
+	if hdr.Number == 0 {
+		return nil
+	}
```

Comment:
This also disables it for real post-merge networks that have a valid post-merge genesis. Instead, I recommend setting `MustBePostMerge: config.Genesis.L2.Number == 0,` in the config, so we can accept pre-merge headers, but only when necessary.

Reply:
But `MustBePostMerge: config.Genesis.L2.Number == 0,` disables the check for all blocks even when the only pre-merge block we get is this genesis. Also I don't think we every pull any pre-merge L1 block

Reply:
Maybe check as `hdr.Number == 0 && config.Genesis.L2.Number != 0` then?

Reply:
There's no reason that a merged-at-genesis chain has to have a difficulty of 0 in its genesis header.  Typically it does and I can see how EIP-3675 says it should be but I'm not aware of any EL that would validate it.  The genesis block just is and can be wildly invalid in a wide range of ways - the only thing that matters is that everyone has the same genesis block.

So I'd actually be ok with skipping checks for all genesis blocks given that we're checking it matches the block hash anyway.